sympy/series/limits.py

```diff
@@ -71,8 +70,16 @@ def heuristics(e, z, z0, dir):
         if r:
             rv = e.func(*r)
             if rv is S.NaN:
-                return
-
+                if e != ratsimp(e):
```

Comment:
Well this will not raise a Polynomial Error as `r` will not be passes in this condition.